pkg/reconciler/dependencybuild/buildrecipeyaml.go

```diff
@@ -507,8 +432,240 @@ mv $(workspaces.source.path)/source/.jbs/build.sh $(workspaces.source.path)`, or
 		}
 	}
 
-	for _, i := range buildTask.Results {
-		ps.Results = append(ps.Results, tektonpipeline.PipelineResult{Name: i.Name, Description: i.Description, Value: tektonpipeline.ResultValue{Type: tektonpipeline.ParamTypeString, StringVal: "$(tasks." + BuildTaskName + ".results." + i.Name + ")"}})
+	if jbsConfig.Spec.ContainerBuilds {
+		fmt.Printf("### Running container builds \n")
+
+		// TODO: ### Note - its also possible to refer to a remote pipeline ref as well as a task.
+		resolver := tektonpipeline.ResolverRef{
+			Resolver: "git",
+			Params: []tektonpipeline.Param{
+				{
+					Name: "url",
+					Value: tektonpipeline.ParamValue{
+						Type:      tektonpipeline.ParamTypeString,
+						StringVal: v1alpha1.KonfluxBuildDefinitions,
+					},
+				},
+				{
+					// TODO: ### Currently always using 'head' of branch.
+					Name: "revision",
+					Value: tektonpipeline.ParamValue{
+						Type:      tektonpipeline.ParamTypeString,
+						StringVal: "main",
+					},
+				},
+				{
+					Name: "pathInRepo",
+					Value: tektonpipeline.ParamValue{
+						Type:      tektonpipeline.ParamTypeString,
+						StringVal: v1alpha1.KonfluxBuildahPath,
+					},
+				},
+			},
+		}
+
+		ps.Tasks = append([]tektonpipeline.PipelineTask{
+			{
+				Name:     BuildTaskName,
+				RunAfter: runAfter,
+				TaskRef: &tektonpipeline.TaskRef{
+					// Can't specify name and resolver as they clash.
+					ResolverRef: resolver,
+				},
+				Timeout: &v12.Duration{Duration: time.Hour * v1alpha1.DefaultTimeout},
+				Params: []tektonpipeline.Param{
+					{
+						Name: "DOCKERFILE",
+						Value: tektonpipeline.ParamValue{
+							Type:      tektonpipeline.ParamTypeString,
+							StringVal: ".jbs/Containerfile"},
+					},
+					{
+						Name: "IMAGE",
+						Value: tektonpipeline.ParamValue{
+							Type:      tektonpipeline.ParamTypeString,
+							StringVal: registryArgsWithDefaults(jbsConfig, buildId)},
+					},
+					{
+						Name: "SOURCE_ARTIFACT",
+						Value: tektonpipeline.ParamValue{
+							Type:      tektonpipeline.ParamTypeString,
+							StringVal: preBuildImage,
+						},
+					},
+				},
+
+				// TODO: ### How to pass build-settings/tls information to buildah task?
+				//       Note - buildah-oci-ta task has no defined workspace
+				//Workspaces: []tektonpipeline.WorkspacePipelineTaskBinding{
+				//	//{Name: WorkspaceBuildSettings, Workspace: WorkspaceBuildSettings},
+				//	{Name: WorkspaceSource, Workspace: WorkspaceSource},
+				//	//{Name: WorkspaceTls, Workspace: WorkspaceTls},
+				//},
```

Comment:
Currently I haven't figured out how (if at all) we can pass the setting.xml/tls shared workspaces - I suspect we'll need a different approach.